lib/block-supports/index.php

```diff
@@ -81,11 +82,11 @@ function gutenberg_apply_block_supports( $block_content, $block ) {
 
 	// Apply new styles and classes.
 	if ( ! empty( $new_classes ) ) {
-		$block_root->setAttribute( 'class', $new_classes );
+		$block_root->setAttribute( 'class', trim( $new_classes ) );
```

Comment:
Just curious why the `trim` was necessary here?  And it may or may not still be needed after rebasing due to recent refactoring #24486

Reply:
It was  necessary because if there were no new styles or classes when   merging with existing styles and classes already on the markup, you'd end up with empty  spaces at the end. Check the diff in the tests expectations where it's pretty clear.

Reply:
Gotcha!  Yeah that makes sense, I think #24486 took care of that as well. 😄